src/main/java/net/mtrop/doom/tools/decohack/data/DEHActionPointer.java

```diff
@@ -134,28 +149,20 @@ public static DEHActionPointer getByMnemonic(String mnemonic)
 	
 	private DEHActionPointer(int frame, String mnemonic)
 	{
-		this(frame, mnemonic, false, 0, 0, 0, 0);
+		this(frame, mnemonic, DEHActionPointerType.DOOM19, new DEHActionPointerParam[0]);
 	}
 
-	private DEHActionPointer(int frame, String mnemonic, boolean mbf)
+	private DEHActionPointer(int frame, String mnemonic, DEHActionPointerType type)
 	{
-		this(frame, mnemonic, mbf, 0, 0, 0, 0);
+		this(frame, mnemonic, type, new DEHActionPointerParam[0]);
 	}
 
-	private DEHActionPointer(int frame, String mnemonic, boolean mbf, int param0min, int param0max)
-	{
-		this(frame, mnemonic, mbf, param0min, param0max, 0, 0);
-	}
-
-	private DEHActionPointer(int frame, String mnemonic, boolean mbf, int param0min, int param0max, int param1min, int param1max)
+	private DEHActionPointer(int frame, String mnemonic, DEHActionPointerType type, DEHActionPointerParam[] params)
```

Comment:
Hah, that's awesome -- I haven't seriously written any Java in like a decade, so I had no idea this was even a thing it could do :P -- will definitely update this.